lib/normalize.js

```diff
@@ -67,31 +67,59 @@ const normalizeSources = dp => {
   for (const idx in dp.sources) {
     dp.sources[idx].title = dp.sources[idx].name
   }
-  for (const idx in dp.resources){
-    for (const id in dp.resources[idx].sources){
-      if(dp.resources[idx].sources[id].title) {
+  for (const idx in dp.resources) {
+    for (const id in dp.resources[idx].sources) {
+      if (dp.resources[idx].sources[id].title) {
         dp.resources[idx].sources[id].title = dp.resources[idx].sources[id].title
       }
-      else if(dp.resources[idx].sources[id].name){
+      else if (dp.resources[idx].sources[id].name) {
         dp.resources[idx].sources[id].title = dp.resources[idx].sources[id].name
       }
       else {
-        dp.resources[idx].sources[id].title = 'unkown'
+        dp.resources[idx].sources[id].title = 'unknown'
       }
     }
   }
   return dp
 }
 
 const normalizeContributors = dp => {
-  for (const idx in dp.contributors){
-    dp.contributors[idx].name = dp.contributors[idx].name.toLowerCase().replace(/\s/g, '_')
+  const normalizeContirutorsTitle = dp => {
+    for (const idx in dp.contributors) {
+      if (dp.contributors[idx].title) {
+        dp.contributors[idx].title = dp.contributors[idx].title
+      } else if (dp.contributors[idx].name) {
+        dp.contributors[idx].title = dp.contributors[idx].name
+        delete dp.contributors[idx].name
+      }
+    }
+  }
+  if (dp.author) {
+    dp.contributors.push({title: dp.author})
+    delete dp.author
+    normalizeContirutorsTitle(dp)
+  } else {
+    normalizeContirutorsTitle(dp)
   }
   return dp
 }
 
+const normalizePath = dp => {
+  for (const idx in dp.resources) {
+    if (dp.resources[idx].path) {
+      if (dp.resources[idx].url) {
+        delete dp.resources[idx].url
+      }
+    }
+    else if (dp.resources[idx].url) {
+      dp.resources[idx].path = dp.resources[idx].url
```

Comment:
Why we're not deleting `url` here? We should either delete or not in both cases. If it does not create bugs in the backend we can keep it but then let's keep it in above code as well.

Reply:
@anuveyatsu will be removed